### DISCUSSION TITLE
Help: PR status update in workflow invoked via comment

### DISCUSSION THREAD

**Discussion by zimventures** [community/community#25141] [Actions]
<p>Howdy everyone :raising_hand_man:</p>
<p>Trying to create a workflow that will run some tests when a PR comment is left. The goal is to have the workflow show up on the PR as a status check as well. </p>
<pre><code>name: Functional Tests

on:
  issue_comment:
    types: [created]
  pull_request_review:
    types: [submitted]

jobs:
  functest:

    # Only run this if the user asked us to
    if: github.event.comment.body == 'run functest'

    runs-on: self-hosted
    container:
      image: python:3.7-buster

    steps:
    - uses: actions/checkout@v2

    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements.txt
    - name: Test with pytest
      run: |
        pip install pytest
        pytest -s basic_functest.py
</code></pre>
<p>In the example above, the workflow is triggered with a comment, however it doesn’t show up as a status check for the PR. </p>
<p>Any help here would be appreciated - thanks! </p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/zimventures">@zimventures</a> ,</p>
<p>It’s not supported. Comment for pull request will not create new commit for the repository. According to the status rest api <a href="https://developer.github.com/v3/repos/statuses/#create-a-status" rel="nofollow noopener">here</a>, you need to have an commit sha firstly, then you can add status to the pull request. Besides, you cannot get the workflow status itself since it’s not completed.</p>
<p>Thanks.</p>

**Comment by zimventures** [comment]
<p>Why can’t it just assume the last commit SHA as the target for the status update? There are certainly plenty of examples out there of actions that perform this type of thing. You’d think it would be allowed directly within a workflow. </p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/zimventures">@zimventures</a> ,</p>
<p>Thanks for your reply!</p>
<p>Yes, as an alternative, you can use the last commit in the pull request as the SHA for status creation. And transfer the test result as ‘success, failure…etc’ to API body.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cae5ecdfc54aa15f1cae877c0f430bdcb0966cd6.png" data-download-href="/uploads/short-url/sWV3P8Od40P3PJ7DcVz3GDU6MUm.png?dl=1" title="addstatus.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181089114-afb00995-d9ca-416b-b389-702d440a3057.png" alt="addstatus.png" data-base62-sha1="sWV3P8Od40P3PJ7DcVz3GDU6MUm" width="690" height="332" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cae5ecdfc54aa15f1cae877c0f430bdcb0966cd6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">addstatus.png</span><span class="informations">768×370 17.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by zimventures** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a> ,</p>
<p>   I’m a little confused as to what you’re suggesting or showing in your example. Could you please provide the source code? It’s unclear if you’re suggesting that this is something that can happen within the workflow of if you’re using the GitHub API directly to add status messages to the PR. </p>
<p>Thanks.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/zimventures">@zimventures</a> ,</p>
<p>This can be completed in the workflow with rest <a href="https://developer.github.com/v3/repos/statuses/#create-a-status" rel="nofollow noopener">api</a>. Code sample as below:</p>
<pre><code>- name: get the one latest&amp;last sha # get the last commit sha in pull_Requst
        run: |
          URL=${{ github.event.issue.pull_request.url }}/commits
          echo "::set-env name=PRSHA::$(curl -u "weide:${{ secrets.GITHUB_TOKEN }}" -s $URL | jq -r '.[] | .sha' | xargs | awk -F " " '{print $NF}')"
      - name: echo PRSHA # check the lastest SHA
        run: |
          echo $PRSHA
      - name: create the status # create the status on the lastest commit sha from pull request
        run: |
          curl -X POST -u "weide:${{ secrets.PAT0309 }}" -H "Content-Type: application/json" "https://api.github.com/repos/${{ github.repository }}/statuses/$PRSHA" --data '{ "state": "failure", "target_url": "https://example.com/build/status", "description": "The build failed!", "context": "continuous-integration/statuscreation7"}'
</code></pre>
<p>Note:</p>
<ol>
<li>
<p>You can transfer the test result as a status (success, failure…) to API body.</p>
</li>
<li>
<p>You need to figure out the latest/last commit SHA in pull request with rest <a href="https://developer.github.com/v3/pulls/#list-commits-on-a-pull-request" rel="nofollow noopener">api</a>, if you use other commit(eg: the 1st commit), the status will be created on the SHA but not added into PR.</p>
</li>
</ol>
<p><img src="https://user-images.githubusercontent.com/52772761/181089117-1d9c591a-f117-4d8a-a829-dab825650751.png" alt="statusadded.png" data-base62-sha1="9xtNAhO4ysoRK9PLgTd0KkWhHoq" width="611" height="244"></p>
<p>Thanks.</p>
